lion/io/database_parameters.h

```diff
@@ -244,4 +257,24 @@ inline void write_parameters(Xml_document& doc, const std::string& path, const s
     }
 }
 
+bool database_parameters_all_used(const Xml_element& element)
+{
+    if(element.has_attribute("__unused__"))
+        if(element.has_children())
+            return database_parameters_all_used(element.get_first_child());
+        else
```

Comment:
Sorry I still don't understand why you need to loop the siblings.

This algorithm should be:

1- If element does not have the attribute -> return false, call is done

2- If element does not have children -> return true, call is done

3- If element has children -> loop children and call `database_parameters_all_used` for all of them. If any children returns false, return false. If you finish looping the children, return true

```C++
bool database_parameters_all_used(const Xml_element& element)
{
    if (  !element.has_attribute("__unused__") )
        return false;

    if ( !element.has_children() )
        return true;

    for (auto child : element.get_children() )
        if ( !database_parameters_all_used(child) )
            return false;

    return true;
}
```

Reply:
Hey! I had chosen the approach because `get_children` function loops through all the siblings and then returns the final vector, which would prove to be inefficient if we are going to iterate over the same set of children after visiting them once during the vector formation. This is why I preferred directly looping over the siblings.

Reply:
Honestly I think that what I sent has got the best trade off between efficiency and readability

Reply:
And I think this is super efficient in the sense that the minimum amount of calls to Tinyxml2 are used. Looping a vector of pointers (Xml_element is a wrapper for a pointer) is free

Reply:
Yup, makes sense